core/app/[locale]/(default)/compare/page-data.ts

```diff
@@ -0,0 +1,85 @@
+import { removeEdgesAndNodes } from '@bigcommerce/catalyst-client';
+import { cache } from 'react';
+
+import { getSessionCustomerAccessToken } from '~/auth';
+import { client } from '~/client';
+import { graphql } from '~/client/graphql';
+import { revalidate } from '~/client/revalidate-target';
+import { ProductCardFragment } from '~/components/product-card/fragment';
+import { getPreferredCurrencyCode } from '~/lib/currency';
+
+const MAX_COMPARE_LIMIT = 10;
```

Comment:
Are we doing anything on the UI as soon as a customer reaches this limit? Thinking maybe we can disable the compare button and/or show a toast?

Reply:
Good idea!

Reply:
![Screenshot 2025-03-18 at 4 01 40 PM](https://github.com/user-attachments/assets/b1e8ce03-0ea5-4948-8a46-322ed9f348ee)

With latest changes, I now alert when the user has met the MAX_COMPARE_LIMIT and disable the checkboxes for unchecked products.